lib/mobile_application_platform/security_token/service.rb

```diff
@@ -16,14 +16,29 @@ def token(application:, icn:)
         Rails.logger.info("#{config.logging_prefix} token success, application: #{application}, icn: #{icn}")
         parse_response(response, application, icn)
       rescue Common::Client::Errors::ClientError => e
-        status = e.status
-        description = e.body.presence && e.body[:error_description]
-        raise e, "#{config.logging_prefix} token failed, client error, status: #{status}," \
-                 " description: #{description}, application: #{application}, icn: #{icn}"
+        parse_and_raise_error(e, icn, application)
       end
 
       private
 
+      def parse_and_raise_error(e, icn, application)
+        status = e.status
+        parse_body = e.body.present? ? parse_json(e.body) : {}
+        context = {
+          error_description: parse_body['error_description'],
+          error: parse_body['error'],
+          error_uri: parse_body['error_uri']
+        }.compact
+        raise e, "#{config.logging_prefix} Token failed, client error, status: #{status}," \
+                 " application: #{application}, icn: #{icn}, context: #{context}"
+      end
+
+      def parse_json(to_parse)
```

Comment:
I probably wouldn't `rescue` an error from `JSON.parse` here, theoretically we should always expect parseable JSON, and if we don't get it, we want to raise an error and get it to Sentry so that we can track it. This is an example of eating the error and removing actionable context